File: 프로그래머스/lv2/12899. 124 나라의 숫자/124 나라의 숫자.js
function solution(n) {
    let temp = n,
        answer = [],
        multiple = n % 3 === 0;

    if (multiple) --temp;
    while (temp !== 0) {
        let remain = parseInt(temp % 3);
        if (remain) answer.unshift(remain);
        else {
            answer.unshift(4);
            temp--;
        }
        temp = parseInt(temp / 3);
    }
    answer = answer.join("");

    if (multiple) answer = answer * 1 + 2;
    return answer.toString();
}

File: 프로그래머스/lv2/12899. 124 나라의 숫자/124 나라의 숫자.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
    string answer = "", str = "412";
    bool three = (n % 3 == 0);
    n -= three ? 1 : 0;

    while (n)
    {
        answer = str[n % 3] + answer;
        if (n % 3 == 0) --n;
        n = n / 3;
    }

    if (three) answer.back() = '4';
    return answer;
}